content/en/docs/features/control-plane/entity-tag.md

```diff
@@ -0,0 +1,151 @@
+
+During device startup, the device developer loads the ETAGs using `oc_etag_load_and_clear` via the public API and clears the existing storage. Additionally, when properly shutting down IoTivity-lite, the ETAGs are stored in the storage using `oc_etag_dump` via the public API. Improper shutdown results in the ETAGs not being stored, leading to their absence upon restart. The device developer should call `oc_etag_load_and_clear` after creating all resources and before the first run by calling `oc_main_poll_v1`, and `oc_etag_dump` before calling `oc_main_shutdown`.
+
+```c
+int ret = oc_main_init(&handler);
+...
+// Create all custom resources
+...
+oc_etag_load_and_clear();
+while (!quit)
+{
+    oc_clock_time_t next_event_mt = oc_main_poll_v1();
+    ...
+}
+oc_etag_dump();
+...
+// Remove all custom resources
+...
+oc_main_shutdown();
+```
+
+### Edge Cases
+
+1. **Time Synchronization**: Ensuring proper time synchronization is crucial for accurately generating the epoch time used in ETAGs. Inconsistent system clock synchronization at device can lead to ETAG conflict between the device and client. To address this, if the epoch time is less than the current ETAG value, the highest ETAG value plus `oc_random_value()+1` is used as a mitigation strategy.
+2. **Storage Failure**: If storing ETAG values fails, it can result in the loss of ETAG information and inconsistencies when determining resource modification. To mitigate this issue, the device developer should verify the return value from `oc_etag_dump` to confirm the success of the operation. Storage failures often occur due to insufficient storage space.
+3. **ETAG Reset**: In the event of a system restart without dump or power outage, the highest ETAG value stored in memory may be lost. To mitigate this, the ETAG values are not loaded through `oc_etag_load_and_clear`, requiring the client to resynchronize with the device.
+4. **Absence of Resources after Device Reboot**: ETAG values for non-existing resources are considered when calculating the highest ETAG value in `oc_etag_load_and_clear`. This ensures that ETAG values are not reused by new resources, avoiding conflicts and maintaining consistency.
+5. **Addition of Resources after Device Reboot**: Similar to creating a new resource, the ETAG value for a newly added resource is set either as the epoch time in milliseconds (`oc_clock_time()`) or as the highest ETAG value `oc_random_value()+1` if the epoch time is less than the current highest ETAG value. This ensures proper tracking of changes for the newly added resource.
+
+## Efficient Device Twin Synchronization
+
+In order to monitor resource changes and determine if a resource has been modified on the device, the CoAP gateway utilizes the Entity Tag (ETAG) mechanism. The Hub maintains the ETAG for each resource along with the timestamp of the last change. It is important to note that the timestamp refers to the time when the notification arrives at the CoAP gateway.
```

Comment:
Why do we need a timestamp?

Reply:
timestamp is needed to determine the latest ETAG. Fixed